libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,370 @@
+            datepickerUigStyle,
+        ];
+    }
+
+    constructor() {
+        super();
+
+        // Fix tot <-> tot-en-met
+        if (Dutch?.nl) {
+            Dutch.nl.rangeSeparator = ' tot en met ';
+        }
+    }
+
+    protected firstUpdated(changedProperties: Map<string, unknown>): void {
+        super.firstUpdated(changedProperties);
+
+        this.initializeComponent();
+
+        if (!this.isInShadowDOM()) {
+            this.getFlatpickrCalendar()?.classList.add('flatpickr-calendar--push-top');
+        }
+
+        if (changedProperties.has('selectedDate')) {
+            if (this.selectedDate) {
+                const defaultDate = this.parseSelectedDate(this.selectedDate);
+                this.updateValue(defaultDate);
+                this.instance?.setDate(defaultDate, false, this.format);
+            }
+        }
+
+        if (!this.initialValue) {
+            this.initialValue = this.value;
+        }
+    }
+
+    updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+
+        const options = this.getDynamicOptions();
+        this.updateOptionsForInstance(options);
+
+        if (changedProperties.has('value')) {
+            this.updateValue(this.value);
+        }
+
+        if (changedProperties.has('block')) {
+            if (this.block) {
+                this.getFlatpickrWrapper()?.classList.add('flatpickr-wrapper--block');
+            } else {
+                this.getFlatpickrWrapper()?.classList.remove('flatpickr-wrapper--block');
+            }
+        }
+    }
+
+    protected render(): TemplateResult {
+        const inputClasses = {
+            'vl-input-field--error': this.error || this.isInvalid,
+            'vl-input-field--success': this.success,
+            'vl-input-field--block': this.block,
+            'vl-input-field--disabled': this.disabled,
+        };
+
+        const buttonClasses = {
+            'vl-input-addon--error': this.error || this.isInvalid,
+            'vl-input-addon--success': this.success,
+            'vl-input-addon--disabled': this.disabled,
+        };
+
+        return html`
+            <div class="vl-input-group" id="wrapper">
+                <input
+                    id=${this.id}
+                    name=${this.name || this.id}
+                    type="text"
+                    class="vl-input-field js-vl-datepicker-input ${classMap(inputClasses)}"
+                    ?required=${this.required}
+                    ?disabled=${this.disabled}
+                    ?error=${this.error}
+                    ?success=${this.success}
+                    ?readonly=${this.readonly}
+                    pattern=${this.pattern}
+                    aria-invalid=${this.error}
+                    aria-label=${this.label}
+                    .value=${live(this.value)}
+                    @input=${this.onInput}
+                />
+                <button
+                    id="button"
```

Comment:
Id weglaten of zinvollere naam geven aub

Reply:
done